lib/parser.ex

```diff
@@ -0,0 +1,205 @@
+defmodule Membrane.Ogg.Parser do
+  import Bitwise
+
+  @type packet :: %{payload: binary, track_id: integer, bos?: boolean, eos?: boolean}
+
+  @spec parse(binary, continued_packets :: %{}) ::
+          {parsed :: list(packet), unparsed :: binary, continued_packets :: %{}}
```

Comment:
* you don't need to name input args in typespec as their names can be read from function definition  
* we need more precise typespec for `continued_packets`

```suggestion
  @spec parse(binary,  %{}) :: {parsed :: [packet], unparsed :: binary, continued_packets :: %{}}
```